games/rpg/src/lib.rs

```diff
@@ -0,0 +1,1483 @@
+                let reward3 = self.system_vault.authorize(||
+                    borrow_resource_manager!(self.token_greavite)
+                        .mint(reward3_d_3));
+                self.system_vault.authorize(|| nft_proof.non_fungible().update_data(nft_data));
+                self.levelup(nft_proof);
+                self.system_vault.put(key_bucket);
+                return (reward1, reward2, reward3)
+            }
+            else {
+                nft_data.exp += exp_enemy_3;
+                let stage_decimal: Decimal = stage.into();
+                nft_data.stage = stage_decimal + dec!(1);
+                let reward1 = self.system_vault.authorize(||
+                    borrow_resource_manager!(self.token_gold)
+                        .mint(reward1_w));
+                let reward2 = self.system_vault.authorize(||
+                        borrow_resource_manager!(self.token_wood)
+                        .mint(reward2_w));
+                let reward3 = self.system_vault.authorize(||
+                    borrow_resource_manager!(self.token_greavite)
+                        .mint(reward3_w));
+                self.system_vault.authorize(|| nft_proof.non_fungible().update_data(nft_data)); 
+                self.levelup(nft_proof);
+                self.system_vault.put(key_bucket);
+                return (reward1, reward2, reward3)
+            }
+        }
+        // Calculates the combat between user and enemy. Enemy Data + rewards set by caller. Returns health after combat
+        // TODO: Add abilities. Add stat changes mid-combat. Add combat vs multiple smaller enemies. Add enemy abilities/gear.
+        pub fn combat(
+            // Kept mutable for now for future implementaion of stat buffs/debuffs mid combat
+            &mut self, 
+            mut health: Decimal, 
+            mut damage: Decimal,
+            mut defense: Decimal,  
+            mut speed: Decimal, 
+            mut enemy_health: Decimal, 
+            mut enemy_damage: Decimal, 
+            mut enemy_defense: Decimal, 
+            mut enemy_speed: Decimal,
+            ) -> Decimal {
+            let speed_ratio = speed / enemy_speed;
+            loop {
+                let mut damage_given = damage - (enemy_defense / dec!(2));
+                if damage_given == dec!(1) || damage_given <= dec!(1) {
+                    damage_given = dec!(1)
+                }
+                let mut damage_taken = enemy_damage  - (defense / dec!(2));
+                if damage_taken == dec!(1) || damage_given <= dec!(1) {
+                    damage_taken = dec!(1)
+                }
+                let mut priority = speed_ratio * self.seed_50();
+                if health < dec!(0) || health == dec!(0) {
+                    return health
+                }
+                if enemy_health < dec!(0) || enemy_health == dec!(0) {
+                    return health
+                }
+                if priority > dec!(1) {
```

Comment:
I think that `priority` should be handled through a rust `loop` instead of having to code in the logic for each possible `priority`.